core/domain/skill_services.py

```diff
@@ -494,21 +487,21 @@ def _save_skill(committer_id, skill, commit_message, change_list):
             'save skill %s: %s' % (skill.id, change_list))
     skill.validate()
 
+    # Skill model cannot be None as skill is passed as parameter here and that
```

Comment:
This is not entirely true. While one creator is working with the skill, another creator can delete it, in which case when the first creator tries saving the skill, the skill wouldn't exist in the datastore, so I'd suggest not removing this check.

Reply:
OK so how can I test it in the backend?

Reply:
You could save a skill, get its ID, delete it and try to update it. Won't that work?

Reply:
Yes I have tried that but update skill won't work in that case. Update_skill calls apply_change_list() and it can't return a deleted skill. See this line here:
https://github.com/oppia/oppia/blob/b9b161ce9a920e75bc901fe5c47978ff778a4a1e/core/domain/skill_services.py#L401

The above line throws an error.

Reply:
Oh, ok, fine then.